command/server/listener.go

```diff
@@ -30,7 +30,7 @@ var BuiltinListeners = map[string]ListenerFactory{
 // NewListener creates a new listener of the given type with the given
 // configuration. The type is looked up in the BuiltinListeners map.
 func NewListener(l *configutil.Listener, logger io.Writer, ui cli.Ui) (net.Listener, map[string]string, reloadutil.ReloadFunc, error) {
-	f, ok := BuiltinListeners[l.Type]
+	f, ok := BuiltinListeners[l.Type.String()]
```

Comment:
Should the **listenerType** be exported (as `ListenerType`) and then the **BuiltinListeners** map could use the **ListenerType** as the key which would mean that the `.String()' would not need to be called.